src/fabric_rpc2.erl

```diff
@@ -96,7 +96,14 @@ changes(DbName, #changes_args{} = Args, StartVector, DbOptions) ->
     changes(DbName, [Args], StartVector, DbOptions);
 changes(DbName, Options, StartVector, DbOptions) ->
     set_io_priority(DbName, DbOptions),
-    #changes_args{dir=Dir} = Args = lists:keyfind(changes_args, 1, Options),
+    #changes_args{dir=Dir, filter=Filter} = Args0 = lists:keyfind(changes_args, 1, Options),
+    Args = case Filter of
+        {fetch, Style, {DbArg, JsonReq, DDocId, FName}} ->
```

Comment:
Also I think we should move this to couch_changes as an API function. I don't like introspecting that filter tuple outside of the module that defines it.